main.cc

```diff
@@ -385,6 +385,14 @@ static auto defer_verbose_shutdown(const char* what, Func&& func) {
                         break;
                     }
                 }
+            } catch (const seastar::abort_requested_exception&) {
+                do_abort = false;
+            } catch (const seastar::broken_semaphore&) {
+                do_abort = false;
+            } catch (const seastar::gate_closed_exception&) {
+                do_abort = false;
+            } catch (const utils::barrier_aborted_exception&) {
+                do_abort = false;
```

Comment:
What should be the reason for abort() then? Why not just _exit() on _any_ exception with a meaningful message then?

(You force-pushd one more entry into this white-list while I was typing my comment and github asked me to refresh the page. This fact makes this comment even more ... relevant :) )

Reply:
The reason for aborting is to sump core to help understand the error.
If it's an error expected on the shutdown path then there's no need for the core dump.
Otherwise, it might be useful.

Reply:
But why are aborted barriers, closed gates and broken semaphores and expected? In particular,

The 1st link
```
INFO  2022-05-23 10:20:54,105 [shard 0] init - Shutting down database
ERROR 2022-05-23 10:20:54,106 [shard 0] init - Unexpected error shutting down database: seastar::named_semaphore_aborted (Semaphore aborted: off-strategy compaction): aborting
Aborting on shard 0.
```
I don't think compaction is expected to hit the aborted semaphore on shutdown, if I'm wrong and it really is, then it should be compaction code to swallow the exception and not propagate it up the stack

The 2nd link
```
INFO  2022-05-23 10:42:13,602 [shard 0] init - Shutting down database
ERROR 2022-05-23 10:42:13,603 [shard 0] init - Unexpected error shutting down database: utils::barrier_aborted_exception (barrier aborted): aborting
Aborting on shard 0.
```
That's clear exception mishandling -- some shard failed to stop the database and aborted the cross-shard-barrier. Another shard stepped on the aborted barrier and threw. The `invoke_on_all(&database::stop)` picked the latter and ignored the former, which _is_ the one to be found

Reply:
I'm not saying we shouldn't fix those cases, but the question is whether a core dump is useful for that.
A backtrace is going to be printed anyhow with this patch and the error message is supposed to cause the dtests to fail.

Let's open issues for the particular flaws you noticed in any case.

Reply:
BTW, I think that both crashes are about the same -- semaphore abort on db stop -- but the latter is just when parallel_for_each picks the "wrong" shard

Reply:
FYI: https://github.com/scylladb/scylla/issues/10666

Reply:
> but the question is whether a core dump is useful for that.

Yup, I agree with that. But I've no idea why it's useless for broken semaphore but useful for, e.g., compaction_stopped_exception (if it anyhow manages to creep up there)